[#181, #180] A few frontend build fixes

lewiscollard opened the pull request:
* Update stylelint-disable lines to use `selector-max-type` and `selector-max-universal` rather than `selector-no-*`.
* Remove a reference to a dead Vue component.
* Force prettier to use single quotes on CSS - it was converting single quotes to double quotes which was causing pre-commits to fail.

Commit:
[#181] Update stylelint disable lines to use `selector-max-type` and `selector-max-type` rather than *-no-type.

Commit:
Force prettier to use single quotes on CSS files.

Commit:
[#180] Remove reference to no-longer-extant Vue component.

dan-gamble commented on {{cookiecutter.repo_name}}/package.json:
```diff
@@ -106,7 +106,7 @@
       ],
       "**/!(definitions)/*.css": [
-        "prettier --write",
+        "prettier --write --single-quote",
```
This should just be `prettier-standard --write` The same as the `*.js` one.

lewiscollard replied:
I thought `prettier-standard` was a Javascript-specific formatter.

dan-gamble replied:
Nah, `prettier-standard` is just `prettier` but with all the rules of StandardJS (Like single quotes)

Commit:
[#179] Fix a lodash compact error.

lewiscollard commented:
Now has a fix for #179.

Commit:
Change command for `prettier` for CSS

https://github.com/onespacemedia/project-template/pull/182#discussion_r141297377

dan-gamble commented:
You could add `git add` to https://github.com/onespacemedia/project-template/blob/4a4c36b23644a9e822aa153772b8afed7adf6c11/%7B%7Bcookiecutter.repo_name%7D%7D/package.json#L115 The Python portion of this. I know that's an outstanding bug

dan-gamble commented:
Looks okay to me though, could be worth starting a project with it and seeing if you can do `yarn run dev` from the get go

Commit:
Do 'git add' for python files at the end of precommit.

Commit:
Fix stylelint error about rules already being disabled.

lewiscollard commented:
@dan-gamble,

> You could add git add to [the] Python portion of this. I know that's an outstanding bug

Done.

> Looks okay to me though, could be worth starting a project with it and seeing if you can do yarn run dev from the get go

Just done so with all of the commits in this PR. `yarn run dev`, `build`, `lint:css`, `lint:js` are all working. :)

dan-gamble commented:
Awesome, if that's the case i'd say this is g2g 👍
